openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ConverterCommonUtils.java

```diff
@@ -223,22 +229,80 @@ public static String getValidName(String identifier) {
         for (AnnotationNode annotation : annotations) {
-            Node annotReference = annotation.annotReference();
-            if (annotReference.toString().trim().equals(annotationReference) && annotation.annotValue().isPresent()) {
-                MappingConstructorExpressionNode listOfAnnotValue = annotation.annotValue().get();
-                for (MappingFieldNode field : listOfAnnotValue.fields()) {
-                    SpecificFieldNode fieldNode = (SpecificFieldNode) field;
-                    if ((fieldNode).fieldName().toString().trim().equals(annotationField)
-                            && fieldNode.valueExpr().isPresent()) {
-                        ExpressionNode expressionNode = fieldNode.valueExpr().get();
-                        return Optional.of(expressionNode.toString().trim().replaceAll("\"", ""));
+            List<String> expressionNode = extractAnnotationFieldDetails(annotationReference, annotationField,
+                    annotation);
+            if (!expressionNode.isEmpty()) {
+                return Optional.of(expressionNode.get(0));
+            }
+        }
+        return Optional.empty();
+    }
+
+    public static List<String> extractAnnotationFieldDetails(String annotationReference, String annotationField,
+                                              AnnotationNode annotation) {
+        List<String> mediaTypes = new ArrayList<>();
+        Node annotReference = annotation.annotReference();
+        if (annotReference.toString().trim().equals(annotationReference) && annotation.annotValue().isPresent()) {
+            MappingConstructorExpressionNode listOfAnnotValue = annotation.annotValue().get();
+            for (MappingFieldNode field : listOfAnnotValue.fields()) {
+                SpecificFieldNode fieldNode = (SpecificFieldNode) field;
+                if (!((fieldNode).fieldName().toString().trim().equals(annotationField)) &&
+                        fieldNode.valueExpr().isEmpty()) {
+                    continue;
+                }
+                ExpressionNode expressionNode = fieldNode.valueExpr().get();
+                if (expressionNode instanceof ListConstructorExpressionNode) {
+                    SeparatedNodeList mimeList = ((ListConstructorExpressionNode) expressionNode).expressions();
+                    for (Object mime : mimeList) {
+                        if (!(mime instanceof BasicLiteralNode)) {
+                            continue;
+                        }
+                        mediaTypes.add(((BasicLiteralNode) mime).literalToken().text().trim().replaceAll("\"", ""));
                     }
+                } else {
+                    mediaTypes.add(expressionNode.toString().trim().replaceAll("\"", ""));
                 }
             }
         }
-        return Optional.empty();
+        return mediaTypes;
     }
 
+    public static List<String> extractAnnotationFieldDetails(String annotationReference, String annotationField,
```

Comment:
Why are we having overloaded util methods with duplicate logic? It seems to me that we can merge these two util methods.

Reply:
Yeah merged it :)